pkgs/nordvpn/default.nix

```diff
@@ -0,0 +1,98 @@
+{ autoPatchelfHook
+, buildFHSEnvChroot
+, dpkg
+, fetchurl
+, lib
+, stdenv
+, sysctl
+, iptables
+, iproute2
+, procps
+, cacert
+, libxml2
+, libidn2
+, zlib
+, wireguard-tools
+}:
+
+let
+  pname = "nordvpn";
+  version = "3.16.2";
+
+  nordVPNBase = stdenv.mkDerivation {
+    inherit pname version;
+
+    src = fetchurl {
+      url =
+        "https://repo.nordvpn.com/deb/nordvpn/debian/pool/main/nordvpn_${version}_amd64.deb";
+      hash = "sha256-6aAslJ2xwj+khF6HOMtkF0iclrUzhBV64xrHgs5Nc2s=";
+    };
+
+    buildInputs = [ libxml2 libidn2 ];
+    nativeBuildInputs = [ dpkg autoPatchelfHook stdenv.cc.cc.lib ];
+
+    dontConfigure = true;
+    dontBuild = true;
+
+    unpackPhase = ''
+      runHook preUnpack
+      dpkg --extract $src .
+      runHook postUnpack
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out
+      mv usr/* $out/
+      mv var/ $out/
+      mv etc/ $out/
+      runHook postInstall
+    '';
+  };
+
+  nordVPNfhs = buildFHSEnvChroot {
+    name = "nordvpnd";
+    runScript = "nordvpnd";
+
+    # hardcoded path to /sbin/ip
+    targetPkgs = pkgs:
+      with pkgs; [
+        nordVPNBase
+        sysctl
+        iptables
+        iproute2
+        procps
+        cacert
+        libxml2
+        libidn2
+        zlib
+        wireguard-tools
+      ];
+  };
+
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin $out/share
+    ln -s ${nordVPNBase}/bin/nordvpn $out/bin
+    ln -s ${nordVPNfhs}/bin/nordvpnd $out/bin
+    ln -s ${nordVPNBase}/share/* $out/share/
+    ln -s ${nordVPNBase}/var $out/
```

Comment:
This feels weird (having `nordVPNBase` and `nordVPNfhs` not as sources but as symlinks), but looks valid to me. :horse_racing: 

Have you tried to "patchelf" `nordvpnd` instead of using a FHS? Probably, you can do everything in a single `mkDerivation` with it.

Reply:
I think this was also suggested in the PRs discussion here https://github.com/NixOS/nixpkgs/pull/220616#discussion_r1176614749 🤔 as long as it works it's fine for me though, unless you feel it's bad :P

Reply:
From best to worse solutions:

auto-patchelf-hook > patchelf > wrapper (LD_LIBRARY_PATH + PATH) > single FHS > symlinking patched with FHS